File: src/2025-03-04-03-28-40--deepseek-chat-0304/python/exercises/practice/wordy/wordy.py
def answer(question):
    if not question.startswith("What is ") or not question.endswith("?"):
        raise ValueError("syntax error")
    
    # Remove "What is " prefix and "?" suffix
    expr = question[8:-1].strip()
    
    # Handle simple number case
    if expr.lstrip('-').isdigit():
        return int(expr)
    
    # Split into tokens
    tokens = expr.split()
    
    # Initialize with first number
    if not tokens[0].lstrip('-').isdigit():
        if any(word.isdigit() for word in tokens):
            raise ValueError("syntax error")
        raise ValueError("unknown operation")
    result = int(tokens[0])
    tokens = tokens[1:]
    
    while tokens:
        # Expect operation then number
        if len(tokens) < 2:
            raise ValueError("syntax error")
        
        op = tokens[0]
        num = tokens[1]
        
        # Handle multi-word operations
        if op == "multiplied" or op == "divided":
            if len(tokens) < 3 or tokens[1] != "by":
                raise ValueError("syntax error")
            num = tokens[2]
            tokens = tokens[1:]  # consume "by"
        
        if not num.lstrip('-').isdigit():
            raise ValueError("syntax error")
        
        num = int(num)
        
        if op == "plus":
            result += num
        elif op == "minus":
            result -= num
        elif op == "multiplied":
            result *= num
        elif op == "divided":
            result //= num
        else:
            raise ValueError("unknown operation")
        
        tokens = tokens[2:]
    
    return result

File: src/2025-03-04-03-28-40--deepseek-chat-0304/javascript/exercises/practice/wordy/wordy.js
export const answer = (question) => {
  // Validate basic question format
  if (!question.startsWith('What is ') || !question.endsWith('?')) {
    throw new Error('Syntax error');
  }

  // Remove wrapper text
  const expr = question.slice(8, -1).trim();

  // Handle empty expression
  if (expr.length === 0) {
    throw new Error('Syntax error');
  }

  // Handle simple number case
  if (/^-?\d+$/.test(expr)) {
    return parseInt(expr, 10);
  }

  // Tokenize the expression
  const tokens = expr.split(/(plus|minus|multiplied by|divided by)/i)
    .map(t => t.trim())
    .filter(t => t.length > 0);

  // Validate token sequence
  if (tokens.length < 3 || tokens.length % 2 === 0) {
    throw new Error('Syntax error');
  }

  // Check for two numbers in a row
  for (let i = 0; i < tokens.length - 1; i++) {
    if (/^-?\d+$/.test(tokens[i]) && /^-?\d+$/.test(tokens[i+1])) {
      throw new Error('Syntax error');
    }
  }

  // Initialize with first number
  let result = parseInt(tokens[0], 10);
  if (isNaN(result)) {
    throw new Error('Syntax error');
  }

  // Process operations left-to-right
  for (let i = 1; i < tokens.length; i += 2) {
    const op = tokens[i].toLowerCase();
    const num = parseInt(tokens[i+1], 10);
    if (isNaN(num)) {
      throw new Error('Syntax error');
    }

    switch(op) {
      case 'plus':
        result += num;
        break;
      case 'minus':
        result -= num;
        break;
      case 'multiplied by':
        result *= num;
        break;
      case 'divided by':
        result = Math.floor(result / num);
        break;
      default:
        throw new Error('Unknown operation');
    }
  }

  return result;
};
